packages/components/src/context-actions/GlobalContextAction.tsx

```diff
@@ -40,8 +50,17 @@ class GlobalContextAction extends Component {
     document.body.removeEventListener('keydown', this.handleKeyDown, true);
   }
 
-  handleContextMenu(e) {
+  handleContextMenu(e: MouseEvent): void {
+    if (!ContextActionUtils.isContextActionEvent(e)) {
+      (e as ContextActionEvent).contextActions = [];
+    }
+
+    if (!ContextActionUtils.isContextActionEvent(e)) {
+      return;
+    }
```

Comment:
What's going on here? The second if statement doesn't make sense after the first one, it'll always be a ContextActionEvent at that point? And then below there's already a spot where it sets the `contextActions` array ... why not just leave that place below and get rid of this?

Reply:
This was a typescript thing

The event param must be a `MouseEvent` to use `addEventListener`. In order to check that `contextActions` existed on the event there's the util function which by returning a type predicate tells TS what the type is. This was just a way to set a default empty array for `contextActions` and then tell TS that e is a ContextActionEvent.

I switched it to something better I think. Uses type assertion (technically there is no "casting" since TS doesn't do any runtime checks), but doesn't need the double call to that method.